sdk/clientcore/core/src/main/java/io/clientcore/core/http/pipeline/HttpPipelineBuilder.java

```diff
@@ -90,20 +131,411 @@ public HttpPipelineBuilder httpClient(HttpClient httpClient) {
+     */
+    public HttpPipelineBuilder setRedirectPolicy(HttpRedirectPolicy redirectPolicy) {
+        this.redirectPolicy = redirectPolicy;
+        policyNames.add(redirectPolicy.getName());
+
+        return this;
+    }
+
+    /**
+     * Sets the {@link HttpRetryPolicy} that the pipeline will use to handle retries.
+     * <p>
+     * If {@code retryPolicy} is null the pipeline will use the default retry policy
+     * ({@link HttpRetryPolicy#HttpRetryPolicy()}.
+     *
+     * @param retryPolicy The retry policy to set.
+     * @return The updated HttpPipelineBuilder object.
+     */
+    public HttpPipelineBuilder setRetryPolicy(HttpRetryPolicy retryPolicy) {
+        this.retryPolicy = retryPolicy;
+        policyNames.add(retryPolicy.getName());
+
+        return this;
+    }
+
+    /**
+     * Sets the {@link HttpCredentialPolicy} that the pipeline will use to authenticate requests.
+     * <p>
+     * If {@code credentialPolicy} is null the pipeline will not authenticate requests.
      *
-     * @param policies Policies to add to the policy set.
+     * @param credentialPolicy The credential policy to set.
+     * @return The updated HttpPipelineBuilder object.
+     */
+    public HttpPipelineBuilder setCredentialPolicy(HttpCredentialPolicy credentialPolicy) {
+        this.credentialPolicy = credentialPolicy;
+        policyNames.add(credentialPolicy.getName());
+
+        return this;
+    }
+
+    /**
+     * Sets the {@link HttpInstrumentationPolicy} that the pipeline will use to instrument HTTP requests and responses.
+     * <p>
+     * If {@code instrumentationPolicy} is null the pipeline will not collect telemetry about HTTP requests and
+     * responses.
      *
+     * @param instrumentationPolicy The telemetry policy to set.
      * @return The updated HttpPipelineBuilder object.
      */
-    public HttpPipelineBuilder policies(HttpPipelinePolicy... policies) {
-        if (pipelinePolicies == null) {
-            pipelinePolicies = new ArrayList<>();
+    public HttpPipelineBuilder setInstrumentationPolicy(HttpInstrumentationPolicy instrumentationPolicy) {
+        this.instrumentationPolicy = instrumentationPolicy;
+        policyNames.add(instrumentationPolicy.getName());
+
+        return this;
+    }
+
+    /**
+     * Adds the given {@code policy} before the {@code position} in the pipeline.
+     * <p>
+     * When this method is called multiple times with the same {@code position}, the policies will be added in the order
+     * they were added. For example, if {@code policy1} and then {@code policy2} are added before
+     * {@link HttpPipelinePosition#RETRY} the resulting ordering will be {@code policy1}, {@code policy2},
+     * {@link HttpRetryPolicy}.
+     * <p>
+     * If the name of the {@code policy} is any of {@code "redirect"}, {@code "retry"}, {@code "credential"},
+     * {@code "logging"}, or {@code "telemetry"} an {@link IllegalStateException} will be thrown. These names are
```

Comment:
Should we remove `logging` from this Javadoc or are we planning to also make it a reserved name?

Reply:
Good catch, this PR predated the merging of logging and telemetry policies and looks like a missed this.